src/test/java/wis/util/InputParserTest.java

```diff
@@ -0,0 +1,111 @@
+package wis.util;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class InputParserTest {
```

Comment:
Comprehensive test cases with comments to describe test details. Good work!

Reply:
Yeah it's quite tedious writing this Junit Test. Some online Regex testers could test this much faster. I feel that Junit is more suitable for testing algorithms, but there's not much algo in this project anyway...